lib/Flow/Operation.php

```diff
@@ -0,0 +1,218 @@
+	}
+
+	public function isAvailableForScope(int $scope): bool {
+		return $scope === FlowManager::SCOPE_USER;
+	}
+
+	/**
+	 * Validates whether a configured workflow rule is valid. If it is not,
+	 * an `\UnexpectedValueException` is supposed to be thrown.
+	 *
+	 * @throws UnexpectedValueException
+	 * @since 9.1
+	 */
+	public function validateOperation(string $name, array $checks, string $operation): void {
+		list($mode, $token) = $this->parseOperationConfig($operation);
+		$this->validateOperationConfig($mode, $token);
+	}
+
+	/**
+	 * Is being called by the workflow engine when an event was triggered that
+	 * is configured for this operation. An evaluation whether the event
+	 * qualifies for this operation to run has still to be done by the
+	 * implementor by calling the RuleMatchers getMatchingOperations method
+	 * and evaluating the results.
+	 *
+	 * If the implementor is an IComplexOperation, this method will not be
+	 * called automatically. It can be used or left as no-op by the implementor.
+	 *
+	 * @since 18.0.0
+	 */
+	public function onEvent(string $eventName, Event $event, IRuleMatcher $ruleMatcher): void {
+		$flows = $ruleMatcher->getMatchingOperations(self::class, false);
+		foreach ($flows as $flow) {
+			try {
+				list($mode, $token) = $this->parseOperationConfig($flow['operation']);
+				$this->validateOperationConfig($mode, $token);
+			} catch(UnexpectedValueException $e) {
+				continue;
+			}
+
+			$room = $this->getRoom($token);
+			$participant = $this->getParticipant($room);
+			$this->chatManager->sendMessage(
+				$room,
+				$participant,
+				'bots',
+				$participant->getUser(),
+				'MESSAGE TODO',
+				new \DateTime(),
+				null
+			);
+		}
+	}
+
+	protected function parseOperationConfig(string $raw): array {
+		/**
+		 * We expect $operation be a json string, containing
+		 * 	't' => string, the room token
+		 *  'm' => int 1..3, the mention-mode (none, yourself, room)
+		 *
+		 * setting up room mentions are only permitted to moderators
+		 */
+
+		$opConfig = \json_decode($raw, true);
+		if(!is_array($opConfig) || empty($opConfig)) {
+			throw new UnexpectedValueException('Cannot decode operation details');
+		}
+
+		$mode = (int)($opConfig['m'] ?? 0);
+		$token = trim((string)($opConfig['t'] ?? ''));
+
+		return [$mode, $token];
+	}
+
+	protected function validateOperationConfig(int $mode, string $token): void {
+		if(($mode < 1 || $mode > 3)) {
+			throw new UnexpectedValueException('Invalid mode');
+		}
+
+		if(empty($token)) {
+			throw new UnexpectedValueException('Invalid token');
+		}
+
+		try {
+			$room = $this->getRoom($token);
+		} catch (RoomNotFoundException $e) {
+			throw new UnexpectedValueException('Room not found', $e->getCode(), $e);
+		}
+
+		if($mode === 3) {
+			try {
+				$participant = $this->getParticipant($room);
+				if (!$participant->hasModeratorPermissions(false)) {
+					throw new UnexpectedValueException('Not allowed to mention room');
```

Comment:
We agreed that we do not want a person be able to spam others with mentions. Now I had the idea that some trusted individuals it might be allowed to trigger mentions to  a crowd, and in Talk they would be identified by Moderators. If it does not make sense, we can kill it. Or extend/relax it later.